src/RegistryCoordinator.sol

```diff
@@ -507,6 +509,12 @@ contract RegistryCoordinator is
         if (operatorId == 0) {
             operatorId = blsApkRegistry.registerBLSPublicKey(operator, params, pubkeyRegistrationMessageHash(operator));
         }
+        if (_operatorInfo[operator].status == OperatorStatus.DEREGISTERED && 
+            params.pubkeyRegistrationSignature.X != 0 &&
+            params.pubkeyRegistrationSignature.Y != 0 
+        ) {
+            operatorId = blsApkRegistry.registerBLSPublicKey(operator, params, pubkeyRegistrationMessageHash(operator));
```

Comment:
won't this call return the hash of the _new_ pubkey? might have to leave the local `operatorId` variable unchanged / explicitly ignore the return value here?

perhaps I'm misremembering something, but I believe the return value of _this_ function (`_getOrCreateOperatorId`) is used in other functions after this is called. 

Or perhaps I'm misinterpreting, and the operator's ID is actually changing here + supposed to be changing?

I guess let me know the intended/desired behavior here, because I'm not quite sure what it is

Reply:
Seems this is indeed changing the operator's ID, so this behavior seems appropriate, assuming changing the ID is the intention 👍